src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -0,0 +1,238 @@
+    }
+
+    @Override
+    public Map<K, V> read(DataInputStream in) throws IOException {
+      int size = Math.min(Math.max(in.readInt(), minSize), maxSize);
+      Map<K, V> map = new HashMap<>(size);
+      for (int i = 0; i < size; i++) {
+        map.put(keyMutator.read(in), valueMutator.read(in));
+      }
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(map);
+    }
+
+    @Override
+    public void write(Map<K, V> map, DataOutputStream out) throws IOException {
+      out.writeInt(map.size());
+      for (Map.Entry<K, V> entry : map.entrySet()) {
+        keyMutator.write(entry.getKey(), out);
+        valueMutator.write(entry.getValue(), out);
+      }
+    }
+
+    @Override
+    protected Map<K, V> makeDefaultInstance() {
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(new HashMap<>(maxInitialSize()));
+    }
+
+    @Override
+    public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
+      int targetSize = prng.closedRange(minInitialSize(), maxInitialSize());
+      Map<K, V> map = underlyingMutableList(reference);
+      map.clear();
+      for (int i = 0; i < targetSize; i++) {
+        initElement(map, prng);
+      }
+    }
+
+    @Override
+    public void mutateInPlace(Map<K, V> reference, PseudoRandom prng) {
+      Map<K, V> map = underlyingMutableList(reference);
+      if (map.isEmpty()) {
+        initElement(map, prng);
+      } else if (!prng.trueInOneOutOf(4)) {
+        // Choose a random entry, and then key or value, to mutate.
+        int i = prng.indexIn(map.size());
+        map.entrySet().stream().skip(i).findFirst().ifPresent(
+            entry -> mutateElement(entry, map, prng));
+      } else {
+        // This will increase the map size up to max and then oscillate around it.
+        // Smaller sizes are already used, so increasing the size should be a valid choice.
+        int currentSize = map.size();
+        if (currentSize < maxSize) {
+          // Create a new entry, as a deep copy could be expensive.
+          initElement(map, prng);
+        } else if (currentSize > minSize) {
+          // Remove a random entry.
+          int i = prng.indexIn(currentSize);
+          map.keySet().stream().skip(i).findFirst().ifPresent(map::remove);
+        } else {
+          // One element map, mutate that one.
+          mutateElement(map.entrySet().iterator().next(), map, prng);
+        }
+      }
+    }
+
+    private void initElement(Map<K, V> map, PseudoRandom prng) {
+      // Lookup an unused key first, as the map may already contain it.
+      // Keys are solely based on the prng and should eventually be unique.
+      K key;
+      do {
+        key = keyMutator.init(prng);
+      } while (map.containsKey(key));
```

Comment:
This is potentially an infinite loop if e.g. the key type is `Boolean`. We should limit the maximum number of attempts. Did you check what fuzztest does in this case?

Reply:
Good point, I didn't think about such narrow types. fuzztest tries 1k times before giving up, if I understand [the code](https://github.com/google/fuzztest/blob/a663ded6c36f050fbdc634a8fc81d553068d71d7/fuzztest/internal/domain.h#L1309) correctly. That seems excessive, doesn't it. Added a retry mechanism.